internal/controller/controllers/agent_controller.go

```diff
@@ -211,6 +212,160 @@ func (r *AgentReconciler) Reconcile(origCtx context.Context, req ctrl.Request) (
 	return r.updateStatus(ctx, log, agent, origAgent, &h.Host, h.ClusterID, nil, false)
 }
 
+func splitNamespacedName(nsName string) (string, string, error) {
```

Comment:
The splitting of namespaced names seems like functionality that could be used more broadly than just in the context of BMH, This function seems more like it should reside somewhere "common"

Reply:
I'm not sure how frequently we're consuming a combined namespace and name string. Also this is usable across the controllers package as it is and I doubt we'd have many places outside of this package that need this kind of logic.